src/pages/auth/authUtils.ts

```diff
@@ -0,0 +1,41 @@
+export type SignupInputId =
+  | "email"
+  | "nickname"
+  | "password"
+  | "passwordConfirmation";
+
+export type LoginInputId =
+  | "email"
+  | "nickname"
+  | "password"
+  | "passwordConfirmation";
+
+export const isValidEmail = (email: string) => {
+  const emailRegex = /^[A-Za-z0-9._%-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$/;
+  return emailRegex.test(email);
+};
+
+export const getErrorMessage = (
+  type: "email" | "nickname" | "password" | "passwordConfirmation",
+  value: string,
+  password?: string
+): string => {
+  const trimmedValue = value.trim();
+  switch (type) {
+    case "email":
+      if (!trimmedValue) return "이메일을 입력해 주세요";
+      if (!isValidEmail(trimmedValue)) return "잘못된 이메일 형식입니다";
+      return "";
```

Comment:
P3:
case 별로 에러 문구를 반환하도록 하신 점 좋습니다~
다만 아래처럼 default error message를 정해주시면 좋지 않을까싶네요~
```suggestion
      return "입력값을 다시 확인해주세요"; // 문구는 일반적인 에러 케이스를 포함할 수 있으면 좋겠습니다.
```